crates/notion-core/src/inventory/serial.rs

```diff
@@ -161,3 +180,197 @@ impl YarnIndex {
         Ok(super::YarnIndex { entries })
     }
 }
+
+// see npm registry API doc:
+// https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PackageMetadata {
+    pub name: String,
+    pub description: Option<String>,
+    pub versions: HashMap<String, PackageVersionInfo>,
+    #[serde(rename = "dist-tags")]
+    pub dist_tags: PackageDistTags,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PackageVersionInfo {
+    // there's a lot more in there, but right now just care about the version
+    #[serde(with = "version_serde")]
+    pub version: Version,
+    pub dist: DistInfo,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PackageDistTags {
+    #[serde(with = "version_serde")]
+    pub latest: Version,
+    pub beta: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct DistInfo {
+    pub shasum: String,
+    pub tarball: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PackageConfig {
+    pub name: String,
+    #[serde(with = "version_serde")]
+    pub version: Version,
+    pub platform: toolchain::serial::Platform,
+    pub bins: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct BinConfig {
+    pub name: String,
+    pub package: String,
+    #[serde(with = "version_serde")]
```

Comment:
This is nice! Could you use this in other places where the intermediate `serial` data structure uses `String` and then calls `Version::parse()` to convert to the real data structure? For example, `serial::NodeEntry` has a bunch of `String` fields that could be `Version` using this technique.

Reply:
good idea - I'll try to use this method where that is the case

Reply:
I moved `version_serde` under the `version` mod, and added `option_version_serde` as well. Both of those are now used in `inventory` and `toolchain` to go directly to and from Version.